timemachine/optimize/protocol.py

```diff
@@ -208,7 +223,142 @@ def max_work_stddev_distance(prev_lam, next_lam, max_step=0.25):
     return max_work_stddev_distance
 
 
-def greedily_optimize_protocol(distance_fxn: DistanceFxn, target_distance=0.5, max_iterations=1000) -> Array:
+# (2) pair overlap
+def reweighted_pair_overlap(u_n_A, u_n_B, u_n_ref):
+    """given arrays [u(x) for x in xs] (for u in {u_A, u_B, u_ref}),
+    estimate overlap(A, B) by reweighting from ref->A and ref->B
+
+    Notes
+    -----
+    * see sec. 3.4 of https://pubmed.ncbi.nlm.nih.gov/25808134/ for expression being approximated
+    * TODO: describe approximation approach
+    """
+    # reduced potentials -> unnormalized log probs
+    log_q_A = -u_n_A
+    log_q_B = -u_n_B
+    log_q_ref = -u_n_ref
+    log_N = jnp.log(len(log_q_ref))
+
+    # pick a normalization constant for ref
+    log_p_ref_n = log_q_ref - logsumexp(log_q_ref - log_N)
+
+    # normalize A and B wrt ref
+    log_p_A = log_q_A - logsumexp(log_q_A - log_p_ref_n - log_N)
+    log_p_B = log_q_B - logsumexp(log_q_B - log_p_ref_n - log_N)
+
+    log_prod_AB_n = log_p_A + log_p_B
+    # assert (log_prod_AB_n < jnp.inf).all()
+
+    log_p_mix_n = logsumexp(jnp.array([log_p_A, log_p_B]), axis=0)
```

Comment:
nit/unsure: is it conventional to define this as

```suggestion
    log_p_mix_n = logsumexp(jnp.array([log_p_A, log_p_B]), axis=0) - jnp.log(2)
```

and omit the final factor of 2 on L262?